compiler.py

```diff
@@ -0,0 +1,230 @@
+    namespace = path[-3]
+    if path[-2] != 'Classes':
+        return
+    if not path[-1].endswith('.uc'):
+        return
+
+    #if path[-1] != 'DXREnemyRespawn.uc':
+    #    return
+
+    content=None
+    with open(file) as f:
+        content = f.read()
+    
+    md5 = hashlib.md5(content.encode('utf-8')).hexdigest()
+
+    #print(file)
+    content_no_comments = strip_comments(content)
+    classline = get_class_line(content_no_comments)
+    # idk if there can be any keywords before the extends/expands keyword? but this regex allows for it
+    # maybe a new keyword could be used since injects doesn't work for DeusExLevelInfo, maybe combines or something
+    injects = re.search(r'class\s+(\S+)\s+(.*\s+)??(((injects)|(extends)|(expands)|(overwrites))\s+([^\s;]+))?', content_no_comments, flags=re.IGNORECASE)
+    classname = None
+    operator = None
+    baseclass = None
+    if injects is not None:
+        classname = injects.group(1)
+        operator = injects.group(4)
+        baseclass = injects.group(9)
+        # if baseclass is not None:
+        #     print("classname: "+classname+", operator: "+operator+", baseclass: "+baseclass)
+        # else:
+        #     print("classname: "+classname+", no inheritance")
+    else:
+        RuntimeError("couldn't read class definition")
+    
+    # maybe do some assertions on classnames, and verify that classname matches filename?
+
+    return {'path':path, 'namespace':namespace, 'filename':filename, 'file':file, 'content':content, 'md5':md5, \
+        'classline':classline, 'classname':classname, 'operator':operator, 'baseclass':baseclass, 'qualifiedclass':namespace+'.'+classname }
+
+
+def proc_file(file, files, injects=None):
+    f = read_uc_file(file)
+    if f is None:
+        return
+    #print("namespace: " + f['namespace'] + ", filename: " + f['filename'] + ", file: " + f['file'] + "\nmd5: " + f['md5'] + "\n" + f['classline'] )
+    if f['operator'] == 'injects':
+        if f['baseclass'] not in injects:
+            injects[f['namespace']+'.'+f['baseclass']] = [ ]
+        injects[f['namespace']+'.'+f['baseclass']].append(f)
+    files[f['qualifiedclass']] = f
+
+
+def write_file(out, f, written, injects):
+    if f['file'] in written:
+        return
+    
+    classname = f['classname']
+    classline = f['classline']
+    content = f['content']
+    
+    if f['qualifiedclass'] in injects:
+        #print(f['qualifiedclass'] + ' has '+ str(len(injects[f['qualifiedclass']])) +' injections, renaming to Base'+f['classname'] )
+        classname = classname+'Base'
+        classline = re.sub('class '+f['classname'], 'class '+classname, classline, count=1)
+    
+    if f['operator'] == 'injects':
+        #print(f['qualifiedclass'] + ' injects into ' + f['baseclass'] )
+        classname = f['baseclass']
+        classline = re.sub('class '+f['classname']+' injects '+f['baseclass'], 'class '+classname+' extends '+f['baseclass']+'Base', classline, count=1)
```

Comment:
we do not yet support a chain of injections, but we would do that just by appending numbers to the class names in the middle of the chain